src/de/mod.rs

```diff
@@ -92,6 +92,32 @@ impl<'de> Deserializer<'de> {
             self.bytes.err(ParseError::TrailingCharacters)
         }
     }
+
+    /// Called from `deserialze_any` when a struct was detected. Decides if there is a
+    /// unit, tuple or usual struct and deserializes it accordingly.
+    ///
+    /// This method assumes there is no identifier left.
+    fn handle_any_struct<V>(&mut self, visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>
+    {
+        // Create a working copy
+        let mut bytes = self.bytes.clone();
```

Comment:
this method doesn't mutate the actual byte stream even on success?

Reply:
It does, but through calling `self.deserialize_tuple` / `self.deserialize_struct`. I need to create a working copy for lookahead first, the bytes do get consumed later.